File: OneToOneCallSample/app/src/test/java/com/stringee/apptoappcallsample/stringee/manager/StringeeCallManagerLogicTest.java
package com.stringee.apptoappcallsample.stringee.manager;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

import com.stringee.apptoappcallsample.stringee.common.ConnectionState;

import org.junit.Test;

public class StringeeCallManagerLogicTest {
    @Test
    public void sameTokenIsIdempotentWhileConnectionIsPending() {
        assertTrue(StringeeCallManager.isSamePendingConnection(
                "token", "token", ConnectionState.CONNECTING));
        assertFalse(StringeeCallManager.isSamePendingConnection(
                "new-token", "token", ConnectionState.CONNECTING));
        assertFalse(StringeeCallManager.isSamePendingConnection(
                "token", "token", ConnectionState.ERROR));
    }

    @Test
    public void reconnectingDisconnectDoesNotLeaveUiInConnectedState() {
        assertTrue(StringeeCallManager.stateAfterDisconnect(true)
                == ConnectionState.CONNECTING);
        assertTrue(StringeeCallManager.stateAfterDisconnect(false)
                == ConnectionState.DISCONNECTED);
    }
}

File: Kotlin-OneToOneCallSample/app/src/test/java/com/stringee/kotlin_onetoonecallsample/stringee/manager/StringeeCallManagerLogicTest.kt
package com.stringee.kotlin_onetoonecallsample.stringee.manager

import com.stringee.kotlin_onetoonecallsample.stringee.common.ConnectionState
import com.stringee.kotlin_onetoonecallsample.stringee.manager.StringeeCallManager.Companion.isSamePendingConnection
import com.stringee.kotlin_onetoonecallsample.stringee.manager.StringeeCallManager.Companion.stateAfterDisconnect
import org.junit.Assert
import org.junit.Test

class StringeeCallManagerLogicTest {
    @Test
    fun sameTokenIsIdempotentWhileConnectionIsPending() {
        Assert.assertTrue(
            isSamePendingConnection(
                "token", "token", ConnectionState.CONNECTING
            )
        )
        Assert.assertFalse(
            isSamePendingConnection(
                "new-token", "token", ConnectionState.CONNECTING
            )
        )
        Assert.assertFalse(
            isSamePendingConnection(
                "token", "token", ConnectionState.ERROR
            )
        )
    }

    @Test
    fun reconnectingDisconnectDoesNotLeaveUiInConnectedState() {
        Assert.assertTrue(
            stateAfterDisconnect(true)
                    == ConnectionState.CONNECTING
        )
        Assert.assertTrue(
            stateAfterDisconnect(false)
                    == ConnectionState.DISCONNECTED
        )
    }
}
